<?xml version="1.0" encoding="UTF-8" ?>
<launch>
        <!-- overwriting these args -->
        <arg name="debug" default="false" />
        <arg name="gui" default="true" />
        <arg name="pause" default="false" />
        <arg name="world" default="sim_world.world" />

        <!-- include gazebo_ros launcher -->
        <include file="$(find gazebo_ros)/launch/empty_world.launch">
                <arg name="world_name" value="$(find surveillance_simulator)/world/$(arg world)" />
                <arg name="debug" value="$(arg debug)" />
                <arg name="gui" value="$(arg gui)" />
                <arg name="paused" value="$(arg pause)" />
                <arg name="use_sim_time" value="true" />
        </include>

        <!-- http://wiki.ros.org/roslaunch/XML/node -->
        <node 
                name="payload_controller" 
                pkg="surveillance_simulator" 
                type="payload_controller.py" 
                respawn="true"
                respawn_delay="3"
        >
                <env name="ROS_PYTHON_LOG_CONFIG_FILE" value="$(find surveillance_simulator)/config/python_logging.conf" />
        </node>

        <node 
                name="stanag_and_payload_connector" 
                pkg="surveillance_simulator" 
                type="stanag_and_payload_connector.py" 
                respawn="true"
                respawn_delay="3"
        >
                <env name="ROS_PYTHON_LOG_CONFIG_FILE" value="$(find surveillance_simulator)/config/python_logging.conf" />
        </node>
</launch>
